### PR TITLE
Netbsd const

### PR DESCRIPTION
**Pull request by vdukhovni**
Rebased #7334 by Christos Zoulas on master, and resolved the "unconst" issue by freeing the dynamic engine ids and names without resort to ENGINE_get_id() and ENGINE_get_name().

I'm leaving the "%z" use in tests alone, we already have lots of those independently of the one added in this commit.

This should be ready for final review merge (assuming Travis and AppVeyor are happy).

### CODE REVIEW

File: test/hmactest.c [hunk 1]
```diff
@@ -27,54 +27,50 @@
...
-        (unsigned char *)"e9139d1e6ee064ef8cf514fc7dc83e86",
+        "e9139d1e6ee064ef8cf514fc7dc83e86",
     },
     {
-        {
-            0x0b, 0x0b, 0x0b, 0x0b, 0x0b, 0x0b, 0x0b, 0x0b, 0x0b, 0x0b, 0x0b,
-            0x0b, 0x0b, 0x0b, 0x0b, 0x0b,
-        }, 16, "Hi There", 8,
-        (unsigned char *)"9294727a3638bb1c13f48ef8158bfc9d",
+	"\x0b\x0b\x0b\x0b\x0b\x0b\x0b\x0b\x0b\x0b\x0b\x0b\x0b\x0b\x0b\x0b",
```
Comment (levitte) [inline test/hmactest.c]: Two more spaces

File: test/hmactest.c [hunk 2]
```diff
@@ -27,54 +27,50 @@
...
         "Jefe", 4, "what do ya want for nothing?", 28,
-        (unsigned char *)"750c783e6ab0b503eaa86e310a5db738",
+        "750c783e6ab0b503eaa86e310a5db738",
     },
     {
-        {
-            0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa,
-            0xaa, 0xaa, 0xaa, 0xaa, 0xaa,
-        }, 16, {
+	"\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa",
```
Comment (levitte) [inline test/hmactest.c]: Here as well

File: test/destest.c [hunk 3]
```diff
@@ -287,7 +287,7 @@ static char *pt(const unsigned char *p, char buf[DATA_BUF_SIZE])
 {
     char *ret;
     int i;
-    static char *f = "0123456789ABCDEF";
+    static const char *f = "0123456789ABCDEF";
```
Comment (paulidale) [inline test/destest.c]: Wouldn't `static const char f[] = "0123456789ABCDEF";` be better?  It avoids a pointer.

File: test/ideatest.c [hunk 4]
```diff
@@ -74,7 +74,7 @@ static int test_idea_cbc(void)
     IDEA_set_encrypt_key(k, &key);
     IDEA_set_decrypt_key(&key, &dkey);
     memcpy(iv, k, sizeof(iv));
-    IDEA_cbc_encrypt((unsigned char *)text, out, text_len, &key, iv, 1);
+    IDEA_cbc_encrypt(text, out, text_len, &key, iv, 1);
```
Comment (levitte) [inline test/ideatest.c]: To make this change, you need to ensure that `text` is an `unsigned char` array

**Comment by levitte** [comment]
Please ensure that all CIs are green before merging

**Comment by vdukhovni** [comment]
> @levitte Please ensure that all CIs are green before merging

Yep, that's the main thing I am waiting for, thanks for the feedback.  I'll squash the fixups.

**Comment by vdukhovni** [comment]
CI passed. Pushed. Thanks!

**Comment by mspncp** [comment]
FWIW: The commits from this pull request were merged as follows:

1057c2c39f Cleaner disposal of ephemeral engine ids and names
9b34028187 Eliminate NOP cast
73ff6d6847 change into hex string constants to avoid overflow warnings
ec91206fbe Add a format attribute to the format functions and fix the broken format strings.
5db120dc35 Fix const issues
bda1f0c0d6 Avoid const castaway warning
d8adf68cd6 Use a const variable
8343229bc4 Use the proper fonst cast
82f518acee don't pass const string where a modifyable string is expected
706a8af015 add missing "void" in prototype.
15f77f2cae add missing const in cast
4b6ae3c3c2 add missing const
f53537b1fa use the proper types to eliminate casts

All commits are missing the "Merged from" annotation, so it' not so easy to find the associated pull request. @openssl/omc wouldn't it be useful to have the server hook reject pushes without "Merged from" annotation, to prevent this type of error?

**Comment by mattcaswell** [comment]
> server hook reject pushes without "Merged from" 

Yes - possibly. Although with some security issues we merge without going via github first (using our private gitlab instance instead). So we'd have to be able to cope with that situation somehow.

**Comment by mspncp** [comment]
How about something like ‚Merged from GitLab xxxx‘? (without hash, which would be interpreted by GitHub)

**Pull request closed without merging**